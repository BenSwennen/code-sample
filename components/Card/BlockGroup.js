import React from "react";
import PropTypes from "prop-types";
import styles from "./Card.css";

const BlockGroup = props => (
  <div className={styles.blockgroup}>
    { props.children }
  </div>
);

BlockGroup.propTypes = {
  children: PropTypes.node,
};

export default BlockGroup;
