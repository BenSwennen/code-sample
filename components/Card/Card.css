$cardBg: $white;
$cardBorderRadius: $borderRadius;

$cardTitleColor: $gray;
$cardTitleSize: .6875rem;
$cardTitleWeight: $fontRegular;

$cardSubtitleColor: $grayLight;
$cardSubtitleSize: 10px;
$cardSubtitleWeight: $fontRegular;

.card {
  background-color: $cardBg;
  border-radius: $cardBorderRadius;
  color: $black;
  display: block;
  padding: $cardContainerPaddingY $cardContainerPaddingX;
  position: relative;

  @mixin cardBoxShadow;

  .block {
    padding: $cardPaddingY $cardPaddingX;
    position: relative;
  }

  > .block {
    &:first-of-type {
      padding-top: 0;
    }

    &:last-of-type {
      padding-bottom: 0;
    }
  }

  .actions {
    float: right;
  }

  .title {
    color: $cardTitleColor;
    display: inline-block;
    font-size: $cardTitleSize;
    font-weight: $cardTitleWeight;
    text-transform: uppercase;
  }
}

.group,
.blockgroup {
  display: flex;
  flex: 0% 1 auto;
  flex-flow: row wrap;
}

.group {
  background: $white;
  border-radius: $borderRadius;

  @mixin cardBoxShadow;

  .card {
    background: transparent;
    border-radius: 0;
    box-shadow: none;

    + .card {
      margin-left: 0;

      @media (--tablet-portrait) {
        border-left: 1px solid $grayLighter;
        box-sizing: border-box;
      }
    }
  }
}
