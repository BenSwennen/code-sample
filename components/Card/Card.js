import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import styles from "./Card.css";

const Card = ({ children, className, onMouseEnter, onMouseLeave }) => (
  <div
    className={cx(styles.card, className)}
    onMouseEnter={onMouseEnter}
    onMouseLeave={onMouseLeave}
  >
    {children}
  </div>
);

Card.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  onMouseEnter: PropTypes.func,
  onMouseLeave: PropTypes.func,
};

export default Card;
