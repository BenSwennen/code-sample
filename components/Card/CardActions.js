import React from "react";
import PropTypes from "prop-types";
import styles from "./Card.css";

const CardActions = ({ children }) => (
  <div className={styles.actions}>
    {children}
  </div>
);

CardActions.propTypes = {
  children: PropTypes.node,
};

export default CardActions;
