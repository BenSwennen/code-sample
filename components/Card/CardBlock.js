import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import styles from "./Card.css";

const CardBlock = ({ className, children }) => (
  <div className={cx(styles.block, className)}>
    {children }
  </div>
);

CardBlock.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

export default CardBlock;
