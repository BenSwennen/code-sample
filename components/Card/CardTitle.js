import React from "react";
import PropTypes from "prop-types";
import styles from "./Card.css";

const CardTitle = props => (
  <div className={styles.title}>{props.title}</div>
);

CardTitle.propTypes = {
  title: PropTypes.string.isRequired,
};

export default CardTitle;
