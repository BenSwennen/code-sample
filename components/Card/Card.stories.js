import React from "react";
import { storiesOf } from "@storybook/react";
import Card, { BlockGroup, CardActions, CardBlock, CardTitle } from ".";
import { Column, Container } from "../Grid";
import Dropdown, { DropdownButton } from "../Dropdown";
import FormButton from "../FormButton/";

const buttonClick = () =>
  console.log("Button clicked"); // eslint-disable-line

const selectInput = selectedInput =>
  console.log(selectedInput); // eslint-disable-line

storiesOf("Card", module)
  .add("Simple", () => (
    <div>
      <Card>
        <CardBlock>
          <CardTitle title="Enquiry Details" />
          <CardActions>
            <FormButton
              label="Save Changes"
              type="save"
              onClick={buttonClick}
            />
            <FormButton
              label="Cancel"
              type="cancel"
              onClick={buttonClick}
            />
          </CardActions>
        </CardBlock>
        <CardBlock>
          This is a card
        </CardBlock>
      </Card>
    </div>
  ))

  .add("Full Example", () => (
    <div>
      <Card>
        <CardBlock>
          <CardTitle title="Enquiry Details" />
          <CardActions>
            <FormButton
              label="Save Changes"
              type="save"
              onClick={buttonClick}
            />
            <FormButton
              label="Cancel"
              type="cancel"
              onClick={buttonClick}
            />
          </CardActions>
        </CardBlock>
        <BlockGroup>
          <Container>
            <Column span="4">
              <CardBlock>
                <fieldset>
                  <label>Size</label>
                </fieldset>
                <fieldset>
                  <label>Location</label>
                  <input type="text" />
                </fieldset>
              </CardBlock>
            </Column>
            <Column span="4">
              <CardBlock>
                <fieldset>
                  <label>Type</label>
                  <Dropdown
                    inputComponent={DropdownButton}
                    inputValueExtractor={v => v}
                    onChange={selectInput}
                    values={["Expansion", "Renewal", "Expiry"]}
                    valueExtractor={v => ({ label: v })}
                    selectedValues={["Expansion"]}
                  />
                </fieldset>
                <fieldset>
                  <label>Reason for moving</label>
                  <input type="text" />
                </fieldset>
              </CardBlock>
            </Column>
            <Column span="4">
              <CardBlock>
                <fieldset>
                  <label>Start Date</label>
                  <input type="date" />
                </fieldset>
                <fieldset>
                  <label>Budget</label>
                  <input type="text" />
                </fieldset>
              </CardBlock>
            </Column>
            <Column span="4">
              <CardBlock>
                <fieldset>
                  <label>Industry</label>
                  <Dropdown
                    inputComponent={DropdownButton}
                    inputValueExtractor={v => v}
                    onChange={selectInput}
                    values={["Health", "Real Estate", "Technology"]}
                    valueExtractor={v => ({ label: v })}
                    selectedValues={["Real Estate"]}
                  />
                </fieldset>
              </CardBlock>
            </Column>
          </Container>
        </BlockGroup>
      </Card>
    </div>
  ));
