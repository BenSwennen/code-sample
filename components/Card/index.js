import Card from "./Card";
import BlockGroup from "./BlockGroup";
import CardActions from "./CardActions";
import CardBlock from "./CardBlock";
import CardGroup from "./CardGroup";
import CardTitle from "./CardTitle";

export default Card;
export { BlockGroup, CardActions, CardBlock, CardGroup, CardTitle };
