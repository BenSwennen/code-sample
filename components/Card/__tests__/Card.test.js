import React from "react";
import renderer from "react-test-renderer";

import Card from "../Card";
import CardActions from "../CardActions";
import CardTitle from "../CardTitle";
import CardBlock from "../CardBlock";

describe("<Card />", () => {
  it("renders a card with children", () => {
    const component = renderer.create(
      <Card>
        <CardBlock>
          <CardTitle title="This is a title" />
          <CardActions />
        </CardBlock>
        <CardBlock>
          This is a card
        </CardBlock>
      </Card>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
