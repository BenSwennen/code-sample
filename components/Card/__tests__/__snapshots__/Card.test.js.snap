// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders a card with children 1`] = `
<div
  className="card"
>
  <div
    className="block"
  >
    <div
      className="title"
    >
      This is a title
    </div>
    <div
      className="actions"
    />
  </div>
  <div
    className="block"
  >
    This is a card
  </div>
</div>
`;
